src/lib/replication/log_store/home_raft_log_store.cpp

```diff
@@ -54,7 +54,9 @@ void HomeRaftLogStore::truncate(uint32_t num_reserved_cnt, repl_lsn_t compact_ls
     auto const start_lsn = start_index();
 
     // compact_lsn will be zero on first time boot, so we should not truncate in that case.
-    if (compact_lsn == 0 || (start_lsn + num_reserved_cnt >= last_lsn)) {
+    // if comapct_lsn > last_lsn, it means compact will/has truncate all logs, so we should not truncate in that case.
+    // FIXME: cannot resolve the concurrency of trucate and comapct (if not first time boot), maybe need lock in future.
+    if (compact_lsn == 0 || compact_lsn > s_cast< repl_lsn_t >(last_lsn) || (start_lsn + num_reserved_cnt >= last_lsn)) {
```

Comment:
Can we just add condition `|| to_store_lsn(compact_lsn)<start_lsn` to simplify line 75~81?

Reply:
We might consider setting truncate_lsn as compact_lsn and adding this condition to simplify the code. However, I'm uncertain if this approach is feasible. In the current implementation, it seems we cannot truncate any log larger than compact_lsn, yet the compact operation will truncate logs. Do you know if there were any prior considerations regarding this?

Otherwise, lines 75-81 in home_raft_log_store are useful in the scenario where `last_lsn > compact_lsn > start_lsn` and` last_lsn - num_reserved_cnt < compact_lsn`.